src/libraries/System.Security.Cryptography.Pkcs/tests/Rfc3161/TimestampTokenTests.cs

```diff
@@ -85,10 +85,20 @@ public static void ParseDocument_ExcessData(string testDataName)
 
                 using (var signerCert = new X509Certificate2(testData.ExternalCertificateBytes))
                 {
-                    // Assert.NoThrow
-                    signedCms.CheckSignature(
-                        new X509Certificate2Collection(signerCert),
-                        true);
+                    if (!SignatureSupport.SupportsRsaSha1Signatures && signerCert.SignatureAlgorithm.FriendlyName == "sha512RSA")
```

Comment:
I'm a little confused by this. We expect it to throw if it does not support SHA1 signatures (so far so good)... and the cert signing algorithm is SHA512RSA. Why would we expect it to throw if the cert algorithm is sha512RSA?

I think one (or all) of the test TST documents is just signed with RSA+SHA1, even if the cert is SHA512.

Maybe the right thing to do is be checking `signedCms.SignerInfos[0].SignatureAlgorithm` for RSAwithSHA1?

Reply:
Right. I was going to ask about this one.

> Maybe the right thing to do is be checking signedCms.SignerInfos[0].SignatureAlgorithm for RSAwithSHA1?

I'll give this a try.